alacritty/src/config/mod.rs

```diff
@@ -156,6 +162,17 @@ fn read_config(path: &PathBuf, cli_config: Value) -> Result<Config> {
     Ok(config)
 }
 
+/// Make sure the values from the config are in a valid range.
+fn validate_config_values(config: &mut Config) {
+    if config.ui_config.window.dimensions.columns_u32() == 1 {
```

Comment:
Deserialization only happens when we use the config. Changing deserialization won't work when the user is setting the dimensions through command line options.

Reply:
CLI options sohuld be handled in the cli.rs file.

Reply:
We want the validation logic to be applied to both cli options and config. Since we can do the validation at one single place I don't think we should repeat it.

Also I'm still confused about how changing the deserialization type helps. std has non-zero type but not non-one types, but we want to allow users to put 0s as the values. The only value we want to forbid is columns being 1.

Reply:
The note about non-zero types is correct and I'm not sure why I suggested them. However a custom deserialization can still be written.

Generally I really don't like introducing another "validation" method here, since it splits the logic into too many different places with no clear structure as to what goes where.

Reply:
The validation method is the cleanest way I can think of to implement this so far. And I think what's happening here is pretty clear actually: we get config from the config file, we override it with cli options, and after the values are final and before the values take effect we "validate".

Letting deserialization do the job would be cleaner if we don't have cli options, but we do, and if we have another "validation" piece for cli we are repeating ourselves in different ways and making things more complicated.

Reply:
It's not clear. It does something that one might want to check in the deserializer directly, but does it later to check CLI options, then uses a "validate" method to ***mutate** something. That's anything but clear.

Reply:
> one might want to check in the deserializer directly

Don't agree with this. One wants to check the final value instead of only what's in the config file, and this check has to happen after we consider cli options. We can do the check in the deserializer, but that leaves a bug that we allow users to give an invalid value through cli options. We can also check the options separately, but by doing that we are repeating ourselves, writing the same logic at 2 different places in 2 different ways. I think that is only going to be uglier.

> a "validate" method to *mutate something

I agree it's bad naming...

Reply:
I'm not sure if it is worth it to check at all tbh, or if we should just `.max(MIN_SIZE)` wherever we need to use it.

Reply:
I think the check should be done eventually. When a user looks at the config file they would assume all values, maybe except for the ones that are so large that the screen cannot hold, can be accepted by the dimensions fields, so it's an annoying surprise and confusion if a certain value is silently ignored. I think we should skip implementing the check in this PR however, so that we have a smaller task to look at.

Reply:
Replicating a validation that **must** be the same in multiple places is a bad approach, clarity aside. It allows divergence in behavior of the config file's constraints from the CLI options, which should never be the case. The question seems to be how to best implement validations in general, since some values fall back to defaults in the parser, and some do not.

> I think we should skip implementing the check in this PR however, so that we have a smaller task to look at.

I think there are configurations unrelated to this PR that also could probably be improved, so I would agree that skipping this and opening a new issue for config/cli option validation/defaults would probably be best. That way we can merge in this feature sooner, and properly think through the validations elsewhere.